src/core/control/Control.cpp

```diff
@@ -2464,6 +2390,85 @@ void Control::fileLoaded(int scrollToPage) {
     updateDeletePageButton();
 }
 
+void Control::promptMissingPdf(LoadHandler& loadHandler, const fs::path& filepath) {
```

Comment:
I have 2 issues here, which are fine for 1.2, as bugfix (nitpick). 
First, the function does more than just prompting for a missing pdf.
Second, the current way is not compatible with gtk4, which is relevant for master.

So, when you have time, you could refactor this after this PR is merged, to accept a handler function, which is called after the dialog has been closed.

It would look then like this:
```suggestion
void Control::promptMissingPdf(LoadHandler& loadHandler, const fs::path& filepath, std::function<void(std::filesystem::path)> handler) {
```

Reply:
The function name will have to change anyways, so I might as well include it here. I propose `attemptReplaceMissingPdf()`, except if you have something better in mind.

Reply:
Actually I'm not sure I understand how this should work with GTK4. We only create and show the dialog, and even if it is modal, GTK will immediately return and call the provided handler upon action from the user?
In this case, why should the handler function already be passed as a parameter to `promptMissingPdf()`? The latter should be able to access the handler function through the class anyways, right?

Would you prefer to omit the name change since it is fitting for GTK4?

Reply:
Yes, just leave the name. 
Gtk3 deprecated old modal dialogs. And Gtk4 removed them. Now you must register a handler function/closure to each dialog, which is called by the event loop, when the dialog closes. 

With c++20 we could just use coroutines and leave the original function structure as is.

Reply:
I see that GTK 4.10 deprecated `GtkMessageDialog` and introduced `GtkAlertDialog`. Can I use `GtkAlertDialog`, or do you want to not have a dependency on 4.10 (at first)?

Reply:
I think we are talking past each other. For now this is fine, since this PR is meant to be merged into 1.2.

On master this would still work, but since we want to make the transition to gtk4. We will have to refactor it. When you have the time and this PR is merged into 1.2, you could do this, since you already touched that code lately.
So for now, just squashing your work is fine here.

Regarding your future work; we first need to remove everything, which got deprecated before GTK 3.24 and which is removed in gtk4.0. This means, we are still on 3.24 and you can't use a dependency from 4.10. GtkAllertDialog won't be available.

Reply:
Ok, good I understand now. Thank you! 
You I squash everything or should I keep the refactoring and the undo handler addition separate?

Reply:
Squashing both together is fine for me.